iceoryx_posh/test/moduletests/test_popo_chunk_receiver.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/internal/mepoo/memory_manager.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/chunk_receiver.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/chunk_receiver_data.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_utils/error_handling/error_handling.hpp"
+#include "iceoryx_utils/internal/posix_wrapper/shared_memory_object/allocator.hpp"
+#include "test.hpp"
+
+#include <memory>
+
+using namespace ::testing;
+
+struct DummySample
+{
+    uint64_t dummy{42};
+};
+
+class ChunkReceiver_testBase : public Test
+{
+  protected:
+    ChunkReceiver_testBase()
+    {
+        m_mempoolconf.addMemPool({CHUNK_SIZE, NUM_CHUNKS_IN_POOL});
+        m_memoryManager.configureMemoryManager(m_mempoolconf, &m_memoryAllocator, &m_memoryAllocator);
+    }
+
+    ~ChunkReceiver_testBase()
+    {
+    }
+
+    void SetUp()
+    {
+    }
+
+    void TearDown()
+    {
+    }
+
+    static constexpr size_t MEMORY_SIZE = 1024 * 1024;
+    uint8_t m_memory[1024 * 1024];
+    static constexpr uint32_t NUM_CHUNKS_IN_POOL = iox::MAX_CHUNKS_HELD_PER_RECEIVER + iox::MAX_RECEIVER_QUEUE_CAPACITY;
+    static constexpr uint32_t CHUNK_SIZE = 128;
+
+    iox::posix::Allocator m_memoryAllocator{m_memory, MEMORY_SIZE};
+    iox::mepoo::MePooConfig m_mempoolconf;
+    iox::mepoo::MemoryManager m_memoryManager;
+
+    iox::popo::ChunkReceiverData m_chunkReceiverData{iox::cxx::VariantQueueTypes::SoFi_SingleProducerSingleConsumer};
+    iox::popo::ChunkReceiver m_chunkReceiver{&m_chunkReceiverData};
+};
+
+class ChunkReceiver_test : public ChunkReceiver_testBase
+{
+  public:
+    ChunkReceiver_test()
+        : ChunkReceiver_testBase()
+    {
+    }
+};
```

Comment:
Why is this additional derivation necessary?

Reply:
Too much copy and waste from other test I guess..